src/components/molecules/movie-reaction-panel/reaction-rate.tsx

```diff
@@ -46,37 +48,33 @@ export const ReactionRate = ({
   hasSeen,
 }: {
   wantToSee?: number | null;
-  movieId?: string;
+  movieId: string;
   hasSeen?: boolean | null;
 }) => {
   const [optimisticHasSeen, setOptimisticHasSeen] = useOptimistic(hasSeen);
   const [optimisticWantToSee, setOptimisticWantToSee] =
     useOptimistic(wantToSee);
   const [_, startTransition] = useTransition();
 
-  const handleReaction = async ({
-    wantToSeeReaction,
-    hasSeenReaction,
-  }: AddOrUpdateReactionPayload) => {
-    if (wantToSeeReaction !== undefined && wantToSeeReaction === wantToSee)
-      return;
+  const handleWantToSeeChange = async (
+    wantToSeeNew: AddOrUpdateReactionPayload["wantToSee"],
+  ) => {
+    startTransition(async () => {
+      setOptimisticWantToSee(wantToSeeNew);
+      await addOrUpdateReaction(movieId, {
+        wantToSee: wantToSeeNew,
+      });
+    });
+  };
 
+  const handleHaveSeenNew = async (
```

Comment:
tu bym tez to nazwal handleHaveSeenChange. bo to jest new tylko za pierwszym razem. albo handleWantToSeeToggle czy cos takiego